archivebox/core/middleware.py

```diff
@@ -35,3 +38,23 @@ def middleware(request):
         return response
 
     return middleware
+
+class ReverseProxyAuthMiddleware(RemoteUserMiddleware):
+    header = 'HTTP_{normalized}'.format(normalized=REVERSE_PROXY_USER_HEADER.replace('-', '_').upper())
+
+    def process_request(self, request):
+        if REVERSE_PROXY_WHITELIST == '':
+            return
+
+        ip = request.META.get('REMOTE_ADDR')
```

Comment:
The name of this header should probably also be a config option. cloudflare, nginx, etc. different reverse proxies set different headers for the user's true origin ip address.

Reply:
This is not the header used by the proxy. This the remote IP address of the caller which is used against whitelist. Only if the IP is included there, the header is taken into consideration. The header itself is set here: https://github.com/ArchiveBox/ArchiveBox/pull/866/files#diff-618efc77bbfa507509fde994d8fa689045cbf17527233bd4d2992ce18d8dcd26R43 and it's configurable via `REVERSE_PROXY_USER_HEADER` config parameter.